Druid/DRAbstractMethodCompilerCompiler.class.st

```diff
@@ -30,37 +30,38 @@ DRCompilerCompiler >> compile [
 ]
 
 { #category : #accessing }
-DRCompilerCompiler >> compilerClass [
+DRAbstractMethodCompilerCompiler >> compilerClass [
 
 	^ compilerClass
 ]
 
 { #category : #accessing }
-DRCompilerCompiler >> compilerClass: anObject [
+DRAbstractMethodCompilerCompiler >> compilerClass: anObject [
 
 	compilerClass := anObject
 ]
 
 { #category : #'generation-IR' }
-DRCompilerCompiler >> configureIRBuilder: generator forMethod: method [
+DRAbstractMethodCompilerCompiler >> configureIRBuilder: generator forMethod: method [
 
 	| startBlock initialBasicBlock |
+
 	"Create the initial basic block"
 	startBlock := generator initialBasicBlock.
 	initialBasicBlock := generator createInitialBasicBlock.
 	startBlock jumpTo: initialBasicBlock.
 
-	generator numberOfArguments: (self numberOfArgumentsFor: method)
+	generator numberOfArguments: method numberOfArguments
```

Comment:
This is why the tests are breaking and there is the problem with the numberOfArguments pragma. This method should be defined on the compiler, because different compilers assume different ways to define number of arguments.

I believe this line should be reverted

Reply:
I don't like this (all implementors of #numberOfArgumentsFor: just use the method passed as parameter to send a message, no single use of the compiler for this...) but I'll revert it anyway.

Reply:
I understand! But in my image I have three implementors of it, and with different implementations.

<img width="439" alt="imagen" src="https://user-images.githubusercontent.com/708322/218891476-7d7f61cb-5d4a-4aeb-8d86-f46df882a641.png">

We can improve it, but I'm saying that probably this change is the cause of the broken tests in this PR.

Reply:
And they are sending a message to the method, yes, but doing different things nevertheless :)